### PR TITLE
add registry metrics

### PR DESCRIPTION
**Pull request by FinalT**
dubbo_register_service_rt_milliseconds_avg
dubbo_register_service_rt_milliseconds_last
dubbo_register_service_rt_milliseconds_max
dubbo_register_service_rt_milliseconds_min
dubbo_register_service_rt_milliseconds_sum

dubbo_registry_directory_num_all
dubbo_registry_directory_num_disable_total
dubbo_registry_directory_num_to_reconnect_total
dubbo_registry_directory_num_valid_total

dubbo_registry_notify_num_last
dubbo_registry_notify_requests_total

dubbo_register_rt_milliseconds_max
dubbo_register_rt_milliseconds_avg
dubbo_register_rt_milliseconds_sum
dubbo_register_rt_milliseconds_min
dubbo_register_rt_milliseconds_last

dubbo_registry_register_requests_succeed_total
dubbo_registry_register_requests_total
dubbo_registry_register_requests_failed_total

dubbo_registry_register_service_total
dubbo_registry_register_service_succeed_total
dubbo_registry_register_service_failed_total
dubbo_registry_subscribe_num_failed_total
dubbo_registry_subscribe_num_succeed_total

dubbo_registry_subscribe_num_total
dubbo_registry_subscribe_service_num_total
dubbo_registry_subscribe_service_num_succeed_total
dubbo_registry_subscribe_service_num_failed_total

dubbo_notify_rt_milliseconds_avg
dubbo_notify_rt_milliseconds_last
dubbo_notify_rt_milliseconds_max
dubbo_notify_rt_milliseconds_min
dubbo_notify_rt_milliseconds_sum

### CODE REVIEW

File: metrics/registry/collector.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+)
+
+func Collector(m metrics.MetricRegistry, c *metrics.ReporterConfig) {
+	rc.regRegistry = m
+}
+
+func init() {
+	metrics.AddCollector("registry_info", Collector)
+	metrics.Subscribe(constant.MetricsRegistry, registryChan)
+	go rc.start()
```
Comment (FoghostCn) [inline metrics/registry/collector.go:41]: I think this init too early ,if metric is not enable, no need to init subscribe and goroutine, maybe move it to collectFunc

File: metrics/registry/metric_set.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	dubboRegNum       = "dubbo_registry_register_metrics_num"
+	dubboRegRt        = "dubbo_registry_register_metrics_rt"
+	dubboRegServerNum = "dubbo_registry_register_server_metrics_num"
+	dubboRegServerRt  = "dubbo_registry_register_server_metrics_rt"
+	dubboNotifyRt     = "dubbo_notify_rt"
+)
+
+var (
+	// register metrics key
+	RegisterMetricRequests        = metrics.NewMetricKey("dubbo.registry.register.requests.total", "Total Register Requests")
```
Comment (FoghostCn) [inline metrics/registry/metric_set.go:54]: "."(dot) is not allowed in prometheus metric name,same below all

File: metrics/registry/collector.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	"time"
+)
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/common/constant"
+	"dubbo.apache.org/dubbo-go/v3/metrics"
+)
+
+var (
+	rc           registryCollector
```
Comment (FoghostCn) [inline metrics/registry/collector.go:30]: I don't see rc variable initialization

**Comment by FinalT** [comment]
Thank you for your comments，I reduced the commits，the new pr is https://github.com/apache/dubbo-go/pull/2366

**Pull request closed without merging**